test/bdd/exec_test.go

```diff
@@ -0,0 +1,334 @@
+	}
+
+	clientSet, err = chaosClient.NewForConfig(config)
+
+	if err != nil {
+		Expect(err).To(BeNil(), "failed to get clientSet")
+	}
+
+	err = v1alpha1.AddToScheme(scheme.Scheme)
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	//Creating crds
+	By("creating chaosengine crd")
+	err = exec.Command("kubectl", "create", "-f", "../../deploy/chaos_crds.yaml").Run()
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	//Creating rbacs
+	err = exec.Command("kubectl", "create", "-f", "../../deploy/rbac.yaml").Run()
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	//Creating Chaos-Operator
+	By("creating operator")
+	err = exec.Command("kubectl", "create", "-f", "../../deploy/operator.yaml").Run()
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	fmt.Println("chaos-operator created successfully")
+
+	//Wait for the creation of chaos-operator
+	time.Sleep(10 * time.Second)
+
+	//Check for the status of the chaos-operator
+	operator, _ := client.CoreV1().Pods("litmus").List(metav1.ListOptions{LabelSelector: "name=chaos-operator"})
+	for _, v := range operator.Items {
+
+		Expect(string(v.Status.Phase)).To(Equal("Running"))
+		break
+	}
+})
+
+//BDD Tests to check secondary resources
+var _ = Describe("BDD on chaos-executor", func() {
+
+	var err error
+	// BDD TEST CASE 1
+	When("Create a test Deployment with nginx image", func() {
+		It("Should create Nginx deployment ", func() {
+			By("building a deployment")
+			//creating nginx deployment
+			deployment := &appv1.Deployment{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "nginx",
+					Namespace: "litmus",
+					Labels: map[string]string{
+						"app": "nginx",
+					},
+					Annotations: map[string]string{
+						"litmuschaos.io/chaos": "true",
+					},
+				},
+				Spec: appv1.DeploymentSpec{
+					Replicas: func(i int32) *int32 { return &i }(3),
+					Selector: &metav1.LabelSelector{
+						MatchLabels: map[string]string{
+							"app": "nginx",
+						},
+					},
+					Template: v1.PodTemplateSpec{
+						ObjectMeta: metav1.ObjectMeta{
+							Labels: map[string]string{
+								"app": "nginx",
+							},
+						},
+						Spec: v1.PodSpec{
+							ServiceAccountName: "litmus",
+							Containers: []v1.Container{
+								{
+									Name:  "nginx",
+									Image: "nginx:latest",
+									Ports: []v1.ContainerPort{
+										{
+
+											ContainerPort: 80,
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			}
+			Expect(err).ShouldNot(
+				HaveOccurred(),
+				"while building deployment nginx in namespace litmus",
+			)
+
+			By("creating above deployment")
+			_, err := client.AppsV1().Deployments("litmus").Create(deployment)
+			Expect(err).To(
+				BeNil(),
+				"while creating deployment nginx in namespace litmus",
+			)
+		})
+	})
+
+	When("Create Pod-delete chaosExperiment in litmus Namespace", func() {
+		It("should create a CustomResource ChaosExperiment Pod-delete", func() {
+
+			By("building a chaosExperiment")
+			ChaosExperiment := &v1alpha1.ChaosExperiment{
```

Comment:
Instead of building it can you get it from the litmus hub

Reply:
Building seems appropriate option for testing purposes

Reply:
If we are building it we have to sync every time from the upstream which is not a good option. cc @ksatchit